pkg/portal/cluster/machines_test.go

```diff
@@ -97,3 +111,148 @@ func TestMachines(t *testing.T) {
 		t.Fatal(r)
 	}
 }
+
+func TestVMAllocationStatus(t *testing.T) {
+	ctx := context.Background()
+	controller := gomock.NewController(t)
+	mockResourcesClient := mock_features.NewMockResourcesClient(controller)
+	mockVirtualMachinesClient := mock_compute.NewMockVirtualMachinesClient(controller)
+	NewResourceClientFunction = func(environment *azureclient.AROEnvironment,
+		subscriptionID string,
+		authorizer autorest.Authorizer) features.ResourcesClient {
+		return mockResourcesClient
+	}
+	NewVirtualMachineClientFunction = func(environment *azureclient.AROEnvironment, subscriptionID string, authorizer autorest.Authorizer) compute.VirtualMachinesClient {
+		return mockVirtualMachinesClient
+	}
+
+	defer func() {
+		NewResourceClientFunction = features.NewResourcesClient
+	}()
+
+	type test struct {
+		name    string
+		mocks   func(*test, *mock_env.MockInterface, *mock_refreshable.MockAuthorizer, *mock_features.MockResourcesClient, *mock_compute.MockVirtualMachinesClient)
+		wantErr string
+	}
+	for _, tt := range []*test{
+		{
+			name: "Everything runs fine",
```

Comment:
The test name could be made a little more descriptive and should clearly state the functionality you're trying to test.

Reply:
Yup .. changed the title, do check and let me know.